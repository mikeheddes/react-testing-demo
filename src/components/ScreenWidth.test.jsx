import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import "jest-styled-components";

import HOCScreenWidth, { ScreenWidth } from "./ScreenWidth";

describe("<ScreenWidth />", () => {
  it("should provide the window.innerWidth as width prop", () => {
    global.window.innerWidth = 456;
    const wrapper = shallow(<HOCScreenWidth />);
    expect(wrapper.find(ScreenWidth).prop("width")).toBe(456);
  });

  it("should update the width on resize event", () => {
    global.window.innerWidth = 567;
    const wrapper = shallow(<HOCScreenWidth />);
    expect(wrapper.find(ScreenWidth).prop("width")).toBe(567);
    global.window.innerWidth = 678;
    // Resize event should be triggerd here
    wrapper.instance().handleResize();
    expect(wrapper.find(ScreenWidth).prop("width")).toBe(678);
  });

  it("should match snapshot", () => {
    const tree = renderer.create(<ScreenWidth width={345} />);
    expect(tree).toMatchSnapshot();
  });
});
