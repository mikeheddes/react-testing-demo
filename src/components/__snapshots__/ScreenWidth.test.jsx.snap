// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenWidth /> should match snapshot 1`] = `
.c0 {
  background-color: #855FEB;
  position: relative;
  box-shadow: 8px 8px rgba(0,0,0,0.15);
}

.c1 {
  display: block;
  text-align: center;
  color: white;
  font-weight: 500;
  font-size: 42px;
  padding: 39px 50px;
}

<div
  className="c0"
  color="#855FEB"
>
  <span
    className="c1"
  >
    345
     px
  </span>
</div>
`;
